kaldb/src/test/java/com/slack/kaldb/testlib/TemporaryLogStoreAndSearcherRule.java

```diff
@@ -80,7 +85,15 @@ public TemporaryLogStoreAndSearcherRule(
             SchemaAwareLogDocumentBuilderImpl.build(
                 fieldConflictPolicy, enableFullTextSearch, metricsRegistry),
             metricsRegistry);
-    logSearcher = new LogIndexSearcherImpl(logStore.getSearcherManager(), logStore.getSchema());
+
+    ConcurrentHashMap<String, LuceneFieldDef> schema = logStore.getSchema();
+
+    // add schema definition for our string property
+    schema.put(
```

Comment:
is adding it here got to do with getting a "known" string field in the schema before we index a document? Because we init the open search adapter with the schema right at the start?

Reply:
This is actually to inform the query of the schema to use, and is unrelated to indexing here. The aggregations use the schema information to correctly build out appropriate aggregation logic (ie, you might do a terms aggregation on either a numeric or text field).